content/rooms/fragments/v1-auth-rules.md

```diff
@@ -26,22 +26,25 @@ The rules are as follows:
         version, reject.
     4.  If `content` has no `creator` field, reject.
     5.  Otherwise, allow.
-2.  Reject if event has `auth_events` that:
-    1.  have duplicate entries for a given `type` and `state_key` pair
-    2.  have entries whose `type` and `state_key` don't match those
+2.  Considering the event's `auth_events`:
+    1.  If there are duplicate entries for a given `type` and `state_key` pair,
+        reject.
+    2.  If there are entries whose `type` and `state_key` don't match those
         specified by the [auth events
         selection](/server-server-api#auth-events-selection)
-        algorithm described in the server specification.
-3.  If event does not have a `m.room.create` in its `auth_events`,
-    reject.
-4.  If the create event content has the field `m.federate` set to `false`
-    and the sender domain of the event does not match the sender domain of
-    the create event, reject.
-5.  If type is `m.room.aliases`:
+        algorithm described in the server specification, reject.
+    3.  If there are entries which were themselves rejected under the [checks
+        performed on receipt of a
+        PDU](server-server-api/#checks-performed-on-receipt-of-a-pdu), reject.
+    4. If there is no `m.room.create` event among the entries, reject.
+    5. If the `m.room.create` event content has the field `m.federate` set to `false`
+       and the `sender` domain of the event does not match the `sender` domain of
+       the create event, reject.
```

Comment:
is this really a check on the `auth_events` or is it a check on the event being inspected?

Reply:
multiplied by however many times it shows up in this PR

Reply:
I don't understand the question. It's a check on the `m.room.create` event which is referred to in the `auth_events` of the event being inspected.

Reply:
shouldn't the server be using the `m.room.create` event from the room rather than `auth_events` though? It feels more understandable to move this point back up a level rather than nest it unmodified into the `auth_events` checks.

Reply:
possibly, but that seems to be a change to its meaning. Currently we have:

3. If event does not have a `m.room.create` in its `auth_events`, reject.
4. If the create event...

It's not entirely clear which create event it is talking about in 4, but given the context from 3, surely it's the one from the `auth_events`.

This is all related to #1136.

Reply:
hrrrm, actually, looking at the source to Synapse, it does appear to apply the test to the create event in the room state, so I guess that's an argument that we should make that explicit.

Reply:
Ok, I've updated this to make it apply to the room state, rather than the auth events.